calculator/main.py

```diff
@@ -37,14 +61,21 @@ def add_new_operation(calc, operation):
     :param operation: Dict with the single operation to be added.
                       ie: {'add': add_function}
     """
-    pass
+    
+    #calc['operations'] = dict(list(calc['operations'].items()) + list(operation.items()))
+    
+    newname = list(operation.keys())[0]
+    newval = operation[newname]
+    calc['operations'][newname] = newval
+    return calc
```

Comment:
Not necessary to return calc at the end of add_new_operation or in reset_history